tests/topotests/multicast_pim6_dual_stack/test_multicast_pim6_dual_stack.py

```diff
@@ -0,0 +1,1936 @@
+        "r1": {"mld": {"interfaces": {intf_r1_i1: {"mld": {"version": "1"}}}}}
+    }
+    result = create_mld_config(tgen, topo, input_dict_1)
+    assert result is True, "Testcase {}: Failed Error: {}".format(tc_name, result)
+
+    step("Enable IGMP on receiver interface on same interface")
+    intf_r1_i2 = topo["routers"]["r1"]["links"]["i2"]["interface"]
+    input_dict_1 = {
+        "r1": {"igmp": {"interfaces": {intf_r1_i2: {"igmp": {"version": "2"}}}}}
+    }
+    result = create_igmp_config(tgen, topo, input_dict_1)
+    assert result is True, "Testcase {}: Failed Error: {}".format(tc_name, result)
+
+    step("Configure RP on FRR2 loopback interface for IPv4 and IPv6 group address")
+    input_dict = {
+        "r2": {
+            "pim": {
+                "rp": [
+                    {
+                        "rp_addr": topo["routers"]["r2"]["links"]["lo"]["ipv4"].split(
+                            "/"
+                        )[0],
+                        "group_addr_range": GROUP_RANGE_4,
+                    }
+                ]
+            },
+            "pim6": {
+                "rp": [
+                    {
+                        "rp_addr": topo["routers"]["r2"]["links"]["lo"]["ipv6"].split(
+                            "/"
+                        )[0],
+                        "group_addr_range": GROUP_RANGE_1,
+                    }
+                ]
+            },
+        }
+    }
+
+    result = create_pim_config(tgen, topo, input_dict)
+    assert result is True, "Testcase {} : Failed Error: {}".format(tc_name, result)
+
+    step("Verify show PIM6 interface traffic without any mld join")
+    intf_r1 = topo["routers"]["r1"]["links"]["r2"]["interface"]
+    state_dict = {"r1": {intf_r1: ["joinRx", "joinTx"]}}
+
+    state_before_v6 = verify_pim_interface_traffic(tgen, state_dict, addr_type="ipv6")
+    assert isinstance(
+        state_before_v6, dict
+    ), "Testcase {} : Failed \n state_before is not dictionary \n " "Error: {}".format(
+        tc_name, result
+    )
+
+    state_before_v4 = verify_pim_interface_traffic(tgen, state_dict, addr_type="ipv4")
+    assert isinstance(
+        state_before_v4, dict
+    ), "Testcase {} : Failed \n state_before is not dictionary \n " "Error: {}".format(
+        tc_name, result
+    )
+
+    step("send mld join (ffaa::1-5) from R1")
+    result = app_helper.run_join("i1", MLD_JOIN_RANGE_1, "r1")
+    assert result is True, "Testcase {}: Failed Error: {}".format(tc_name, result)
+
+    step("Send IGMP joins from DUT for group range 225.1.1.1-5")
+    input_join = {
+        "i2": topo["routers"]["i2"]["links"]["r1"]["interface"],
+    }
+
+    for recvr, recvr_intf in input_join.items():
+        result = app_helper.run_join(recvr, IGMP_JOIN_RANGE_1, join_intf=recvr_intf)
+        assert result is True, "Testcase {}: Failed Error: {}".format(tc_name, result)
+
+    step(
+        "IPv4 mld and IPv6 MLD join received on R1 from same interface"
```

Comment:
is it the same interface?

Reply:
yes it is same interface, but updated comment as below:

IPv4 igmp and IPv6 mld join received on R1 from same interface